pkg/kubectl/cmd/drain.go

```diff
@@ -612,19 +614,22 @@ func (o *DrainOptions) evictPods(pods []corev1.Pod, policyGroupVersion string, g
 		globalTimeout = o.Timeout
 	}
 	globalTimeoutCh := time.After(globalTimeout)
+	numPods := len(pods)
 	for {
 		select {
-		case err := <-errCh:
-			return err
-		case <-doneCh:
+		case err := <-returnCh:
 			doneCount++
-			if doneCount == len(pods) {
-				return nil
+			if err != nil {
+				errors = append(errors, err)
 			}
 		case <-globalTimeoutCh:
 			return fmt.Errorf("Drain did not complete within %v", globalTimeout)
 		}
+		if doneCount == numPods {
```

Comment:
This condition could have been part of the "for" statement:  `for doneCount < numPods { ... }`.